docs/datamodel/links.rst

```diff
@@ -475,11 +475,134 @@ in these cases due to better ergonomics of selecting, updating, and even
+query. If the outer person ("Bob" in the example) already exists and only the
+links need to be added, this can be done in an ``update`` query.
+
+Updating a link property is similar except you can select the link instead of
+selecting from the object type since the link has already been established.
+Here, we've discovered that Alice is actually Bob's step-sister, so we want to
+change the link property on the already-established link between the two:
+
+.. code-block:: edgeql
+
+  update Person {
+    name := "Bob",
+    family_members := (
+      select .family_members {
+        @relationship := "step-sister"
+      }
+      filter .name = "Alice"
+    )
+  }
+
+.. warning::
+
+    A link property cannot be referenced in a set union *except* in the case of
+    a :ref:`for loop <ref_eql_for>`. That means this will *not* work:
+
+    .. code-block:: edgeql
+
+        # 🚫
+        insert Movie {
+          title := 'The Incredible Hulk',
+          actors := {(
+              select Person {
+                @character_name := 'The Hulk'
+              } filter .name = 'Mark Ruffalo'
+            ),
+            (
+              select Person {
+                @character_name := 'Abomination'
+              } filter .name = 'Tim Roth'
+            )}
+        };
+
+    That query will produce an error: ``QueryError: invalid reference to link
+    property in top level shape``
+
+    You can use this workaround instead:
+
+    .. code-block:: edgeql
+
+        # ✅
+        insert Movie {
+          title := 'The Incredible Hulk',
+          actors := assert_distinct((
+            with characters := {
+              ('The Hulk', 'Mark Ruffalo'),
+              ('Abomination', 'Tim Roth')
+            },
+            for character in characters union (
+              select Person {
+                @character_name := character.0
+              } filter .name = character.1
+            )
+          ))
+        };
+
+    Note that we are also required to wrap the ``actors`` query with
+    :eql:func:`assert_distinct` here to assure the compiler that the result set
+    is distinct.
+
+
+Querying link properties
+^^^^^^^^^^^^^^^^^^^^^^^^
+
+To query a link property, add the link property's name prepended with ``@`` to
+a shape on the link.
+
+.. code-block:: edgeql-repl
+
+  edgedb> select Person {
+  .......   name,
+  .......   family_members: {
+  .......     name,
+  .......     @relationship
+  .......   }
+  ....... };
+  {
+    default::Person {name: 'Alice', family_members: {}},
+    default::Person {
+      name: 'Bob',
+      family_members: {
+        default::Person {name: 'Alice', @relationship: 'step-sister'}
+      }
+    },
+  }
+
+.. note::
+
+    In the query results above, Alice appears to have no family members even
+    though we know that, if she is Bob's step-sister, he must be her
+    step-brother. We would need to update Alice manually before this is
+    reflected in the database.
```

Comment:
Question for whoever reviews: Is this necessary? When I was reading this, I noticed that the `family_members` link on `Alice` was empty, and for just a second, it hit me as weird until my brain turned the corner to, "of course it's blank because the database doesn't know anything about reciprocal familial relationships." Thought it would be good to just briefly explain, but maybe it's silly. It feels a little out of place, but also it hits weird for a split second when you see it without a note.

Reply:
I think it's good to err on the side of extra info as you've done here.

(And actually this could be a good place to mention that link properties cannot be required if you want to add it to this PR)

Reply:
I ended up mentioning this as an additional note at the top under the note about "link links" vs. link properties. (I feel like this is kind of a place for general caveats, so it made sense to include both here. We've just told you what they are, but here's what you should know before you get any deeper.) I also reiterated it at the very end in the note about the familial relationship only being reflected in the direction you explicitly added the link to reinforce.